### PR TITLE
ID-11

### PR DESCRIPTION
**Pull request by SharmaDinesh**

### CODE REVIEW

File: app/screens/claim/index.tsx [hunk 1]
```diff
@@ -1,21 +1,61 @@
-﻿﻿import React from "react";
-import { Text, View } from "react-native";
+﻿﻿import React, {useState, useEffect} from "react";
```
Comment (bitcoinbrisbane) [inline app/screens/claim/index.tsx:1]: Not to nit pick, but...

```suggestion
﻿﻿import React, { useState, useEffect } from "react";
```

File: app/screens/claim/index.tsx [hunk 2]
```diff
@@ -1,21 +1,61 @@
...
 import styles from "../../styles";
 import EmailClaim from "./EmailClaim";
-import * as DocumentPicker from "expo-document-picker";
 import { useRootStore } from "../../store/rootStore";
 import { observer } from "mobx-react-lite";
 import DateClaim from "./DateClaim";
 import OtherClaim from "./OtherClaim";
 import MobileClaim from "./MobileClaim";
 import SelectClaim from "./SelectClaim";
+import * as ImagePicker from 'expo-image-picker';
```
Comment (bitcoinbrisbane) [inline app/screens/claim/index.tsx:11]:
```suggestion
import * as ImagePicker from "expo-image-picker";
```

File: app/screens/claim/index.tsx [hunk 3]
```diff
@@ -1,21 +1,61 @@
...
+import * as DocumentPicker from "expo-document-picker";
+import AsyncStorage from "@react-native-async-storage/async-storage";
 
 const Claim = () => {
+  useEffect(() => {
+    (async () => {
+      if (Platform.OS !== 'web') {
```
Comment (bitcoinbrisbane) [inline app/screens/claim/index.tsx:18]:
```suggestion
      if (Platform.OS !== "web") {
```

File: app/screens/claim/index.tsx [hunk 4]
```diff
@@ -1,21 +1,61 @@
...
+        const { status } = await ImagePicker.requestMediaLibraryPermissionsAsync();
+        if (status !== 'granted') {
```
Comment (bitcoinbrisbane) [inline app/screens/claim/index.tsx:20]:
```suggestion
        if (status !== "granted") {
```

File: app/screens/claim/index.tsx [hunk 5]
```diff
@@ -1,21 +1,61 @@
...
+          alert('Sorry, we need camera roll permissions to make this work!');
```
Comment (bitcoinbrisbane) [inline app/screens/claim/index.tsx:21]:
```suggestion
          alert("Sorry, we need camera roll permissions to make this work!");
```

File: app/screens/claim/index.tsx [hunk 6]
```diff
@@ -1,21 +1,61 @@
...
+    try {
+      let result = await ImagePicker.launchImageLibraryAsync({
+        mediaTypes: ImagePicker.MediaTypeOptions.All,
+        allowsEditing: false,
+        aspect: [4, 3],
+        quality: 1,
+      })
+      console.log(result)
+      if (!result.cancelled) {
+        await AsyncStorage.setItem('library_url', result.uri);
```
Comment (bitcoinbrisbane) [inline app/screens/claim/index.tsx:52]:
```suggestion
        await AsyncStorage.setItem("library_url", result.uri);
```

File: app/screens/claim/index.tsx [hunk 7]
```diff
@@ -1,21 +1,61 @@
...
   const uploadFile = async () => {
-    const res = await DocumentPicker.getDocumentAsync({});
-    console.log(res);
-    //TODO choose files to upload and save to local storage or local database
+    try {
+      let result = await DocumentPicker.getDocumentAsync({
+        type: '*/*'
+      });
+   
+      if (result.type === 'success') {
```
Comment (bitcoinbrisbane) [inline app/screens/claim/index.tsx:33]:
```suggestion
      if (result.type === "success") {
```

File: app/screens/claim/index.tsx [hunk 8]
```diff
@@ -1,21 +1,61 @@
...
+        await AsyncStorage.setItem('document_url', result.uri);
```
Comment (bitcoinbrisbane) [inline app/screens/claim/index.tsx:34]:
```suggestion
        await AsyncStorage.setItem("document_url", result.uri);
```

**Review by bitcoinbrisbane — CHANGES_REQUESTED** [review]
Just some linting issues

File: app/screens/claim/index.tsx (hunk 1 above)
Comment (SharmaDinesh) [reply app/screens/claim/index.tsx:1]: Done

File: app/screens/claim/index.tsx (hunk 2 above)
Comment (SharmaDinesh) [reply app/screens/claim/index.tsx:11]: Done

File: app/screens/claim/index.tsx (hunk 3 above)
Comment (SharmaDinesh) [reply app/screens/claim/index.tsx:18]: Done

File: app/screens/claim/index.tsx (hunk 4 above)
Comment (SharmaDinesh) [reply app/screens/claim/index.tsx:20]: Done

File: app/screens/claim/index.tsx (hunk 5 above)
Comment (SharmaDinesh) [reply app/screens/claim/index.tsx:21]: Done

**Review by bitcoinbrisbane — APPROVED** [review]
LGTM

File: app/screens/claim/index.tsx [hunk 9]
```diff
@@ -1,21 +1,61 @@
...
+      if (Platform.OS !== "web") {
+        const { status } = await ImagePicker.requestMediaLibraryPermissionsAsync();
+        if (status !== "granted") {
+          alert("Sorry, we need camera roll permissions to make this work!");
+        }
+      }
+    })();
+  }, []);
+
   const uploadFile = async () => {
```
Comment (ahinchliff) [inline app/screens/claim/index.tsx:27]: this shouldnt be called upload if you are storing it to local stoage.

File: app/screens/claim/index.tsx [hunk 10]
```diff
@@ -1,21 +1,61 @@
...
-    const res = await DocumentPicker.getDocumentAsync({});
-    console.log(res);
-    //TODO choose files to upload and save to local storage or local database
+    try {
+      let result = await DocumentPicker.getDocumentAsync({
+        type: "*/*"
+      });
+   
+      if (result.type === "success") {
+        await AsyncStorage.setItem("document_url", result.uri);
```
Comment (ahinchliff) [inline app/screens/claim/index.tsx:34]: does this mean you can only store one document at a time? Wont you need to store documents for each claim?

**Review by ahinchliff — CHANGES_REQUESTED** [review]

File: app/screens/claim/index.tsx (hunk 9 above)
Comment (SharmaDinesh) [reply app/screens/claim/index.tsx]: Change function name UploadFile -> UploadFileFromBrowser

File: app/screens/claim/index.tsx [hunk 11]
```diff
@@ -1,21 +1,74 @@
...
+  const uploadFileFromBrowser = async () => {
+    try {
+      const libraryUrl = await AsyncStorage.getItem("library_url");
+      if(libraryUrl !== null) {
+        let result = await DocumentPicker.getDocumentAsync({
+          type: "*/*"
+        });
+     
+        if (result.type === "success") {
+          console.log('result.uri', result.uri);
```
Comment (xiangzhe0812) [inline app/screens/claim/index.tsx:36]: Can we remove console.log?

File: app/screens/claim/index.tsx [hunk 12]
```diff
@@ -1,21 +1,74 @@
...
+          await AsyncStorage.setItem("document_url", result.uri);
+        }
+      } else {
+        console.log("You can select one document at a time");
```
Comment (xiangzhe0812) [inline app/screens/claim/index.tsx:40]: what's the logic?  just console.log?

File: app/screens/claim/index.tsx [hunk 13]
```diff
@@ -1,21 +1,74 @@
...
+    try {
+      const documentUrl = await AsyncStorage.getItem("document_url");
+      if(documentUrl !== null) {
+        let result = await ImagePicker.launchImageLibraryAsync({
+          mediaTypes: ImagePicker.MediaTypeOptions.All,
+          allowsEditing: false,
+          aspect: [4, 3],
+          quality: 1,
+        })
+        console.log(result)
```
Comment (xiangzhe0812) [inline app/screens/claim/index.tsx:59]: remove console.log

File: app/screens/claim/index.tsx [hunk 14]
```diff
@@ -1,21 +1,74 @@
...
+        if (!result.cancelled) {
+          await AsyncStorage.setItem("library_url", result.uri);
+        }
+      }else {
+        console.log("You can select one document at a time");
```
Comment (xiangzhe0812) [inline app/screens/claim/index.tsx:64]: same above. whats the logic?

**Review by ahinchliff — APPROVED** [review]

**Review by xiangzhe0812 — APPROVED** [review]

**Pull request merged**